src/ovs_stats.c

```diff
@@ -0,0 +1,982 @@
+                    "a61b7e2b-6951-488a-b4c6-6e91343960b2"
+                  ],
+                  [
+                    "iface-status",
+                    "active"
+                  ]
+                ]
+              ]
+          }
+        }
+      }
+    }
+   */
+  const char *path[] = {"Interface", NULL};
+  yajl_val ports = yajl_tree_get(jupdates, path, yajl_t_object);
+  pthread_mutex_lock(&g_stats_lock);
+  if (ports && YAJL_IS_OBJECT(ports))
+    for (int i = 0; i < YAJL_GET_OBJECT(ports)->len; i++)
+      ovs_stats_update_iface(YAJL_GET_OBJECT(ports)->values[i]);
+  pthread_mutex_unlock(&g_stats_lock);
+  return;
+}
+
+/* Handle JSON with Interface table initial values */
+static void ovs_stats_interface_table_result_cb(yajl_val jresult,
+                                                yajl_val jerror) {
+  if (YAJL_IS_NULL(jerror))
+    ovs_stats_interface_table_change_cb(jresult);
+  else
+    ERROR("%s: Error received from OvSDB. Table: Interface", plugin_name);
+  return;
+}
+
+/* Setup OVS DB table callbacks  */
+static void ovs_stats_initialize(ovs_db_t *pdb) {
+  const char *bridge_columns[] = {"name", "ports", NULL};
+  const char *port_columns[] = {"name", "interfaces", NULL};
+  const char *interface_columns[] = {"name", "statistics", "_uuid",
+                                     "external_ids", NULL};
+
+  /* subscribe to a tables */
+  ovs_db_table_cb_register(pdb, "Bridge", bridge_columns,
+                           ovs_stats_bridge_table_change_cb,
+      ovs_stats_bridge_table_result_cb,
+                           OVS_DB_TABLE_CB_FLAG_INITIAL |
+                           OVS_DB_TABLE_CB_FLAG_INSERT |
+          OVS_DB_TABLE_CB_FLAG_MODIFY);
+
+  ovs_db_table_cb_register(pdb, "Bridge", bridge_columns,
+                           ovs_stats_bridge_table_delete_cb, NULL,
+                           OVS_DB_TABLE_CB_FLAG_DELETE);
+
+  ovs_db_table_cb_register(pdb, "Port", port_columns,
+                           ovs_stats_port_table_change_cb,
+      ovs_stats_port_table_result_cb,
+                           OVS_DB_TABLE_CB_FLAG_INITIAL |
+                           OVS_DB_TABLE_CB_FLAG_INSERT |
+          OVS_DB_TABLE_CB_FLAG_MODIFY);
+
+  ovs_db_table_cb_register(pdb, "Port", port_columns,
+                           ovs_stats_port_table_delete_cb, NULL,
+                           OVS_DB_TABLE_CB_FLAG_DELETE);
+
+  ovs_db_table_cb_register(pdb, "Interface", interface_columns,
+                           ovs_stats_interface_table_change_cb,
+      ovs_stats_interface_table_result_cb,
+                           OVS_DB_TABLE_CB_FLAG_INITIAL |
+                           OVS_DB_TABLE_CB_FLAG_INSERT |
+          OVS_DB_TABLE_CB_FLAG_MODIFY);
+}
+
+/* Check if bridge is configured to be monitored in config file */
+static int ovs_stats_is_monitored_bridge(const char *br_name) {
+  int rc = 0;
+  /* if no bridges are configured, return true */
+  if (!(rc = (g_monitored_bridge_list_head == NULL)))
```

Comment:
It's a bit hard to follow this code. I believe if you can get rid of rc which you are treating as a bool this can be simplified a bit.